### ISSUE TITLE
openapi-react-query with null values

### ISSUE THREAD

**Issue by HugeLetters** [openapi-ts/openapi-typescript#1973]
**Description**

openapi-react-query crashes if endpoint returns nulls.

**Reproduction**

any endpoint which returns null with useQuery exposed by this library

**Expected result**

null is both valid json and react-query allows to return it form queryFn
https://github.com/openapi-ts/openapi-typescript/blob/781cf926288d6f735096437b529cc20d51142ccf/packages/openapi-react-query/src/index.ts#L144
This line causes the issue - `!data` should be replaced with `data!==undefined`

**Checklist**

- [ ] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/openapi-ts/openapi-typescript/blob/main/packages/openapi-fetch/CONTRIBUTING.md))

**Comment by kerwanp** [comment]
Hi, happy to accept a PR to fix this issue!

**Comment by BenJeau** [comment]
Affected by the same issue whenever a `useMutation` data response is just an empty string

**Comment by agrahamg** [comment]
This is also happening for me in mutations for endpoints returning a 204
```
  /example:
    post:
      parameters: []
      responses:
        '204':
          description: 'No content'
```

The `fetchClient` returns undefined which errors in the mutation.

**Comment by martmalo** [comment]
Also encountered this issue on http status 204. React query mutation throws undefined error.

**Comment by HugeLetters** [comment]
@martmalo what does the mutation supposed to return? Is it like a `null` or an empty string?

Since I'm working on the PR I would like to know if I need to handle some extra case - right now the fix is that we allow to return anything from mutation except `undefined`. `undefined` is allowed in mutations by react-query but it's not valid JSON so it's not like you can get it as a response.

**Comment by martmalo** [comment]
> @martmalo what is the mutation supposed to return? Is it like a `null` or an empty string?
> 
> 
> 
> Since I'm working on the PR I would like to know if I need to handle some extra case - right now the fix is that we allow to return anything from mutation except `undefined`. `undefined` is allowed in mutations by react-query but it's not valid JSON so it's not like you can get it as a response. 
> 
> 
> 
> You can check what it's supposed to return in devtools in the network tab by checking the associated request

I think you already handled my case because 204 by default is no content response and the response body is undefined.

**Comment by gt-vs-3141** [comment]
@HugeLetters, should we be able to retrieve undefined responses from a mutation? When the API responds with 204, the body is `undefined,` so currently, the mutation doesn't support successful responses without a body definition. 

I think throwing for `undefined` data is a mistake since you mentioned that react-query itself supports it. Types might need to be changed to correctly support it but a type error is preferable over a runtime error.

**Comment by HugeLetters** [comment]
@gt-vs-3141 yeah, thanks for pointing this out - I will adjust the PR to allow undefined in mutations but there might be a small downside that it won't be reflected in the types, mutation data will be of type `never` in these cases but I don't think that's a big deal since undefined response already implies you're not supposed to use it.
